bdk_core/src/sparse_chain.rs

```diff
@@ -180,15 +185,17 @@ impl SparseChain {
             }
         }
 
-        for (tx_height, txid) in &update.txid_by_height {
+        for (tx_index, txid) in &update.indexed_txids {
             // ensure all currently confirmed txs are still at the same height (unless, if they are
             // to be invalidated, or originally unconfirmed)
-            if let Some(&old_height) = self.txid_to_index.get(txid) {
-                if old_height < TxHeight::Confirmed(invalid_from) && tx_height != &old_height {
+            if let Some(old_tx_index) = self.txid_to_index.get(txid) {
+                if old_tx_index.height < TxHeight::Confirmed(invalid_from)
+                    && tx_index.height != old_tx_index.height
```

Comment:
This should be comparing the index. You can't change the index either without a reorg (e.g. position in block, or timestamp etc).